### PR TITLE
Adds QDEL_LIST helpers

### PR DESCRIPTION
**Pull request by Cyberboss**

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1279,6 +1279,8 @@ proc/pick_closest_path(value, list/matches = get_fancy_list_of_atom_types())
 
 #define QDEL_IN(item, time) addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, item), time, TIMER_STOPPABLE)
 #define QDEL_NULL(item) qdel(item); item = null
+#define QDEL_LIST(L) for(var/I in L) qdel(I); L.Cut()
```
Comment (coiax) [inline code/__HELPERS/unsorted.dm]: Can you change this to `if(L) ...` in case the list is lazy?

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -1279,6 +1279,8 @@ proc/pick_closest_path(value, list/matches = get_fancy_list_of_atom_types())
...
+#define QDEL_LIST_ASSOC(L) for(var/I in L) qdel(L[I]); L.Cut()
```
Comment (RemieRichards) [inline code/__HELPERS/unsorted.dm]: but what about ```list(MOB = ATOM)```??

**Comment by MrPerson** [comment]
Isn't it faster to just do list = null and let it GC rather than Cut it? I mean strictly speaking it's not as safe, but 99% of the time, this is just going to be for deleting a locally referenced list that's not been passed around.

**Comment by Cyberboss** [comment]
Where that is a possibility it should be done, but for shit that needs to
run destroy this is what it must do.

**Comment by optimumtact** [comment]
feature freze

**Pull request merged**